src/transmission.js

```diff
@@ -126,27 +177,38 @@ export default class Transmission {
       }
     };
 
-    eachPromise(batch, (ev) => {
-      var url = urljoin(ev.apiHost, "/1/events", ev.dataset);
+    let batches = Object.keys(batchAgg.batches).map((k) => batchAgg.batches[k]);
+    eachPromise(batches, (batch) => {
+      var url = urljoin(batch.apiHost, "/1/batch", batch.dataset);
       var req = superagent.post(url);
 
+      let { encoded, numEncoded } = batchAgg.encodeBatchEvents(batch.events);
```

Comment:
Added code to use it such that if `numEncoded === 0`, we know the batch failed to encode anything.  Avoids sending anything to shepherd at all (it would have sent an empty array).